db/db_impl/db_impl_open.cc

```diff
@@ -698,6 +704,25 @@ Status DBImpl::Recover(
   return s;
 }
 
+Status DBImpl::VerifySstUniqueIdInManifest() {
+  mutex_.AssertHeld();
+  ROCKS_LOG_INFO(
+      immutable_db_options_.info_log,
+      "Verifying SST unique id between MANIFEST and SST file table properties");
+  Status status;
+  for (auto cfd : *versions_->GetColumnFamilySet()) {
+    if (!cfd->IsDropped()) {
+      auto version = cfd->current();
+      version->Ref();
+      mutex_.Unlock();
+      status = version->VerifySstUniqueIds();
```

Comment:
Will errors be swallowed?

Reply:
ping

Reply:
> I'm surprised ASSERT_STATUS_CHECKED didn't catch anything

I am just passing by and don't know the full details but IIRC a problem was the below code would fail if we enforced a `Status` was checked before copy-assignment:

```
Status s; // OR `Status s = Status::OK();`
if (...) {
  s = ...;
} else {
  s = ...;
}
```